txstratum/protocol.py

```diff
@@ -156,6 +196,25 @@ def start_periodic_tasks(self) -> None:
             self.refresh_job_task = Periodic(self.refresh_job, self.JOB_UPDATE_INTERVAL)
             asyncio.ensure_future(self.refresh_job_task.start())
 
+        if self.messages_timeout_task is None:
+            # Start messages timeout periodic task.
+            self.messages_timeout_task = Periodic(self.messages_timeout_job, self.MESSAGES_TIMEOUT_INTERVAL)
+            asyncio.ensure_future(self.messages_timeout_task.start())
+
+    async def messages_timeout_job(self) -> None:
+        """Period task to check for messages that have timed out."""
+        now = txstratum.time.time()
+
+        should_delete = []
+
+        for message in self.messages_in_transit.values():
+            if now > message.timeout:
+                should_delete.append(message.id)
+
+        for msgid in should_delete:
+            message = self.messages_in_transit.pop(msgid)
```

Comment:
With this timeout we might never have the client version if it times out?

Should we have a retry mechanism for it? 

We are using the version "only" for saving metrics in a better way, right?

Reply:
> With this timeout we might never have the client version if it times out?
The timeout happens after 30 seconds, which is a lot of time.

I created it just to make sure we wouldn't keep them in memory.
We can even increase the time.

If the miner really didn't respond in this time, then we surely had some other problem that would happen even if we waited more time.

> Should we have a retry mechanism for it?
> 
> We are using the version "only" for saving metrics in a better way, right?

Yeah, only for metrics. And if we can't get the version for some reason, we will just get an `unknown` in Prometheus tags.

I think we are safe enough in this case that a retry isn't that important.
But maybe in the future, if we send more critical messages using this mechanism.

Reply:
Yeah, I agree